VContainer/Assets/VContainer/Runtime/Internal/TypeAnalyzer.cs

```diff
@@ -206,6 +206,7 @@ public static InjectTypeInfo Analyze(Type type)
             var injectMethods = default(List<InjectMethodInfo>);
             var injectFields = default(List<FieldInfo>);
             var injectProperties = default(List<PropertyInfo>);
+            var startType = type;
```

Comment:
Here is the bug (or misunderstanding) which I fixed. The thing is `type` after while while recursive circle became `System.Object` for all classes. It is because type rewrites in 244-245 line as BaseType.